### ISSUE TITLE
[bug] missing CMakeUserPresets.json

### ISSUE THREAD

**Issue by magras** [conan-io/conan#11286]
Conan 1.48.0 does not generate `CMakeUserPresets.json` if `conanfile.txt` used instead of `.py`.

Probably this is duplicate of #11172.

### Environment Details
  * Operating System: linux
  * Conan version: 1.48.0

### Steps to reproduce
```
cd "$(mktemp -d)"
conan new hello/0.1 -m=cmake_exe
rm conanfile.py
echo -e "[generators]\nCMakeToolchain" > conanfile.txt
mkdir build
(cd build && conan install ..)
tree "$(pwd)"
```

### Output
```
/tmp/tmp.6FdEpSMGTR
├── CMakeLists.txt
├── build
│   ├── CMakePresets.json
│   ├── conan.lock
│   ├── conan_toolchain.cmake
│   ├── conanbuildinfo.txt
│   ├── conaninfo.txt
│   └── graph_info.json
├── conanfile.txt
├── include
│   └── hello.h
├── src
│   ├── hello.cpp
│   └── main.cpp
└── test_package
    └── conanfile.py
```

**Comment by franramirez688** [comment]
Hi @magras 

That's expected because it needs a ``layout`` to create the ``CMakeUserPresets.json``, so it's impossible to do that without a ``conanfile.py``:

```python
    def layout(self):
        cmake_layout(self)
```

That's why it's failing when a ``conanfile.txt`` is there instead.
More information [here](https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmaketoolchain.html#using-the-toolchain-in-developer-flow).

**Comment by magras** [comment]
> That's expected because it needs a `layout` to create the `CMakeUserPresets.json`

So what I should do if I'm using `.txt` and `CMakeToolchain` generator in my projects?
* As a short term solution I guess I can symlink  `CMakePresets.json` to the root of the source dir.
* What will be the long term solution? Rewrite `.txt` to `.py`?

**Comment by franramirez688** [comment]
I guess it could be better to rewrite it and try a simple one. For instance, this could be good enough:

```python
from conan import ConanFile
from conan.tools.cmake import CMake, cmake_layout


class PkgConan(ConanFile):
    settings = "os", "compiler", "build_type", "arch"
    generators = "CMakeToolchain"

    def layout(self):
        cmake_layout(self)

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()

    def package(self):
        cmake = CMake(self)
        cmake.install()
```
If you only want to use that generator, then I think you can use this one. The advantages of using it are that you can always customize it, e.g., if you need more ad-hoc configurations, builds, etc.

UPDATED: even simpler:

```python
from conan import ConanFile
from conan.tools.cmake import cmake_layout


class PkgConan(ConanFile):
    settings = "os", "compiler", "build_type", "arch"
    generators = "CMakeToolchain"

    def layout(self):
        cmake_layout(self)
```

**Comment by magras** [comment]
IIRC toolchain generator should become the main generator for cmake in conan 2.0.
Does it mean that usage of `.txt` is discouraged and will be deprecated in 2.0?

**Comment by memsharded** [comment]
> Does it mean that usage of .txt is discouraged and will be deprecated in 2.0?

No, the ``.txt`` will still be there. And the new CMake integration will still generate a ``CMakePresets.json`` file, it is only the ``CMakeUserPresets.json`` that might be generated under certain conditions (it is not already existing), and that might be used to aggregate multiple different configurations (each one might have a different ``CMakePresets.json`` in a different folder), but for this functionality to be manageable, a ``layout`` is necessary, otherwise it becomes too messy. So it is the ``CMakeUserPresets.json`` that only will be generated for more advanced cases that will require a ``conanfile.py``, while for simple usage a ``conanfile.txt`` with the ``CMakePresets.json`` would be the solution.

**Comment by magras** [comment]
So again what is the long term solution for `.txt`?
* to use symlink or to write my own `CMakeUserPresets.json` that includes `build/CMakePresets.json`?
  It sounds reasonable for a dev environment, but I dislike it in most other scenarios, such as on a build server.
* to ask cmake devs for a new option to specify preset file location?

**Comment by memsharded** [comment]
Of course, requesting the possibility to CMake to allow defining a custom location of the presets file would be ideal. I don't understand why they won't allow specifying the location of a presets file. 

Besides that, yes, you have different possibilities, the one you commented. Or just use a ``conanfile.py``, for 98% of usage it is irrelevant if .py or .txt, because it is just used, and for the other cases, a simple .py is practically as trivial as the conanfile.txt.

We could re-check if it would make sense to generate a CMakeUserPresets.json too for that case, but maybe it won't, lets see what @lasote thinks.

**Comment by memsharded** [comment]
We have discussed this, and this requires some changes that could be breaking in 1.X, so we are moving this to 2.0 scope:

The changes are:
- Change the default ``layout``, so the build and generators folders do not pollute the current directory, something like ``build/generators``
- Make the ``CMakeToolchain`` generate ``CMakeUserPresets.json``, also for ``conanfile.txt``

In the meantime, please use the ``CMakePresets`` (copy or link it), or move to a ``conanfile.py``. Thanks!

**Comment by magras** [comment]
I think that minimizing the difference in behavior between `.txt` and `.py` will lead to less confusion among the users. I appreciate your decision. Thank you.

**Comment by lasote** [comment]
We have finally implemented https://github.com/conan-io/conan/pull/11348 to be released in Conan 1.49. So you can declare:

```
[layout]
cmake_layout

```
in the ``conanfile.txt`` to get the ``CMakeUserPresets.json`` generated correctly
